libraries/aws_ecrpublic_repositories.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsEcrPublicRepositories < AwsResourceBase
+  name 'aws_ecrpublic_repositories'
+  desc 'Verifies settings for a collection of AWS ECR Public Repositories'
+
+  example "
+    describe aws_ecrpublic_repositories do
+      its('count') { should eq 3 }
+    end
+
+    describe aws_ecrpublic_repositories.where(name: 'public-repo') do
+      it { should exist }
+    end
+  "
+
+  attr_reader :table
+
+  FilterTable.create
+             .register_column(:arns, field: :arn)
+             .register_column(:names, field: :name)
+             .register_column(:uris, field: :uri)
+             .register_column(:created_at_dates, field: :created_at)
+             .register_column(:image_tag_mutability_status, field: :image_tag_mutability)
+             .register_column(:image_scanning_on_push_status, field: :image_scanning_on_push_status)
+             .install_filter_methods_on_resource(self, :table)
+
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters(allow: %i(registry_id))
+    @query_params = {}
+    # Validate registry_id. (Optional. If not provided, AWS account ID will be used by the AWS API.)
+    if opts.key?(:registry_id)
+      raise ArgumentError, "#{@__resource_name__}: `registry_id` should be a string of 12 digits." unless /^[0-9]{12}$/.match?(opts[:registry_id])
+      @query_params[:registry_id] = opts[:registry_id]
+    end
+    @table = fetch_data
+  end
+
+  def fetch_data
+    ecrpublic_repositories_rows = []
+    @query_params[:max_results] = 1000
+    loop do
+      catch_aws_errors do
+        @api_response = @aws.ecrpublic_client.describe_repositories(@query_params)
+      end
+      return [] if !@api_response || @api_response.empty?
+
+      @api_response.repositories.each do |repo|
+        ecrpublic_repositories_rows += [{
+          arn: repo.repository_arn,
+          name: repo.repository_name,
+          uri: repo.repository_uri,
+          created_at: repo.created_at,
+        }]
+      end
+      break unless @api_response.next_token
+      query_params[:next_token] = @api_response.next_token
```

Comment:
shouldn't this be an instance variable?
```suggestion
      @query_params[:next_token] = @api_response.next_token
```

Reply:
I left it that way because the resource is similar to ecr_repositories which has the same. Also, i did not think much about it because the integration test for it was successful.

![int_test_ECRPublic](https://user-images.githubusercontent.com/30339457/114399260-015f6100-9ba1-11eb-9440-8585f0691283.png)

Reply:
I guess the integration is passing because most probably there isn't anything above 1000 results.. might have been a bug with `ecr_repo` as well!

Reply:
Yea right.. in that case, i will commit the change